orchestrator/relayer/src/fee_manager.rs

```diff
@@ -0,0 +1,219 @@
+use ethereum_gravity::{
+    utils::GasCost,
+};
+use ethers::prelude::*;
+use ethers::types::Address as EthAddress;
+use gravity_utils::ethereum::{format_eth_address};
+use gravity_utils::types::{ Erc20Token, };
+use std::collections::HashMap;
+use std::time::Duration;
+use std::time::Instant;
+use gravity_utils::types::config::RelayerMode;
+use reqwest::Error;
+
+pub struct FeeManager {
+    token_price_map: HashMap<String, String>,
+    token_api_url_map: HashMap<String, String>,
+    next_batch_send_time: HashMap<EthAddress, Instant>,
+    mode: RelayerMode,
+}
+
+#[derive(serde::Deserialize, Debug)]
+struct ApiResponse {
+    status: String,
+    result: ApiResult,
+}
+
+#[derive(serde::Deserialize, Debug)]
+struct ApiResult {
+    tokenPairDecimal: u32,
+    aggregatedPrice: f64,
+}
+
+impl FeeManager {
+    pub async fn new_fee_manager(mode: RelayerMode) -> FeeManager {
+        let mut fm =  Self {
+            token_price_map: Default::default(),
+            token_api_url_map: Default::default(),
+            next_batch_send_time: HashMap::new(),
+            mode
+        };
+
+        let success = fm.init()
+            .await;
+
+        if !success {
+            panic!("Could not initialize fee manager");
+        }
+        return fm;
+    }
+
+    async fn init(&mut self) -> bool {
```

Comment:
maybe we can return `Result` instead of bool? then we can use `?` directly instead of checking the `bool` when use `init`